File: humanlayer/core/approval_resolve_backend_test.py
import pytest

from humanlayer import ApprovalMethod, CloudHumanLayerBackend, HumanLayer
from humanlayer.testing import env_var


def test_no_args() -> None:
    with env_var("HUMANLAYER_API_KEY", ""):
        hl = HumanLayer()
        assert hl.approval_method == ApprovalMethod.CLI


def test_cli_hardcoded() -> None:
    hl = HumanLayer(approval_method=ApprovalMethod.CLI)
    assert hl.approval_method == ApprovalMethod.CLI


def test_cloud() -> None:
    with env_var("HUMANLAYER_API_KEY", ""):
        with pytest.raises(Exception) as e:
            HumanLayer.cloud()
        assert "HUMANLAYER_API_KEY is required for cloud approvals" in str(e.value)


def test_cloud_endpoint_kwarg_default() -> None:
    hl = HumanLayer(api_key="foo")
    assert hl.approval_method == ApprovalMethod.BACKEND
    assert hl.backend is not None
    assert isinstance(hl.backend, CloudHumanLayerBackend)
    assert hl.backend.connection.api_key == "foo"
    assert hl.backend.connection.api_base_url == "https://api.humanlayer.dev/humanlayer/v1"


def test_cloud_endpoint_kwarg() -> None:
    hl = HumanLayer(api_key="foo", api_base_url="fake")
    assert hl.approval_method == ApprovalMethod.BACKEND
    assert hl.backend is not None
    assert isinstance(hl.backend, CloudHumanLayerBackend)
    assert hl.backend.connection.api_base_url == "fake"


def test_env_var_cloud() -> None:
    with env_var("HUMANLAYER_API_KEY", "foo"):
        hl = HumanLayer()
        assert hl.approval_method == ApprovalMethod.BACKEND
        assert hl.backend is not None
        assert isinstance(hl.backend, CloudHumanLayerBackend)
        assert hl.backend.connection.api_key == "foo"
        assert hl.backend.connection.api_base_url == "https://api.humanlayer.dev/humanlayer/v1"


def test_timeout() -> None:
    """
    test that we can configure the timeout
    """
    with (
        env_var("HUMANLAYER_HTTP_TIMEOUT_SECONDS", "30"),
        env_var("HUMANLAYER_API_KEY", "foo"),
    ):
        hl = HumanLayer()
        assert hl.backend is not None
        assert True  # removed http timeout test

File: humanlayer-ts/src/approval_resolve_backend.test.ts
import { ApprovalMethod, humanlayer, HumanLayer } from './approval'
import { with_env_var, without_env_var } from '../src/testing/index'
import { CloudHumanLayerBackend } from './cloud'

test('no args', () => {
  with_env_var('HUMANLAYER_API_KEY', '', () => {
    const hl = humanlayer()
    expect(hl.approvalMethod).toBe(ApprovalMethod.cli)
  })
})

test('cli hardcoded', () => {
  const hl = humanlayer({ approvalMethod: ApprovalMethod.cli })
  expect(hl.approvalMethod).toBe(ApprovalMethod.cli)
})

test('HumanLayer.cloud()', () => {
  with_env_var('HUMANLAYER_API_KEY', '', () => {
    expect(() => HumanLayer.cloud()).toThrow('HUMANLAYER_API_KEY is required for cloud approvals')
  })
})

test('HumanLayer.cloud() with apiKey', () => {
  without_env_var('HUMANLAYER_API_BASE', () => {
    const hl = HumanLayer.cloud({ apiKey: 'abc' })
    expect(hl.approvalMethod).toBe(ApprovalMethod.backend)
    expect(hl.backend).toBeDefined()
    expect(hl.backend).toBeInstanceOf(CloudHumanLayerBackend)
    const cloudBackend = <CloudHumanLayerBackend>hl.backend
    expect(cloudBackend.connection.apiKey).toBe('abc')
    expect(cloudBackend.connection.apiBaseURL).toBe('https://api.humanlayer.dev/humanlayer/v1')
  })
})

test('Humanlayer with api key and endpoint', () => {
  const hl = new HumanLayer({ apiKey: 'abc', apiBaseUrl: 'fake' })
  expect(hl.approvalMethod).toBe(ApprovalMethod.backend)
  expect(hl.backend).toBeDefined()
  expect(hl.backend).toBeInstanceOf(CloudHumanLayerBackend)
  const cloudBackend = <CloudHumanLayerBackend>hl.backend
  expect(cloudBackend.connection.apiKey).toBe('abc')
  expect(cloudBackend.connection.apiBaseURL).toBe('fake')
})

test('HumanLayer() with env var', () => {
  without_env_var('HUMANLAYER_API_BASE', () => {
    with_env_var('HUMANLAYER_API_KEY', 'abc', () => {
      const hl = new HumanLayer()
      expect(hl.approvalMethod).toBe(ApprovalMethod.backend)
      expect(hl.backend).toBeDefined()
      expect(hl.backend).toBeInstanceOf(CloudHumanLayerBackend)
      const cloudBackend = <CloudHumanLayerBackend>hl.backend
      expect(cloudBackend.connection.apiKey).toBe('abc')
      expect(cloudBackend.connection.apiBaseURL).toBe('https://api.humanlayer.dev/humanlayer/v1')
    })
  })
})
